handshake_ix.go

```diff
@@ -73,206 +73,174 @@ func ixHandshakeStage0(f *Interface, vpnIp uint32, hostinfo *HostInfo) {
 
 }
 
+// Note we always return `false` in ixHandshakeStage1 because there is nothing
+// to tear down.
 func ixHandshakeStage1(f *Interface, addr *udpAddr, hostinfo *HostInfo, packet []byte, h *Header) bool {
 	var ip uint32
-	if h.RemoteIndex == 0 {
-		ci := f.newConnectionState(false, noise.HandshakeIX, []byte{}, 0)
-		// Mark packet 1 as seen so it doesn't show up as missed
-		ci.window.Update(1)
+	if h.RemoteIndex != 0 {
+		// We already have this index, add the new remote addr
+		f.hostMap.AddRemote(ip, addr)
```

Comment:
I'm a bit concerned by this zero value `ip`, is that right?

Reply:
hmmm yeah good point, looking at the blame this has been wrong for a long time! This must be dead code then that never actually triggers.